### PR TITLE
Merge login lib changes and implement new interface methods 

### PR DESCRIPTION
**Pull request by AmandaRiu**
This PR merges the `woocommerce-android` changes to the login library from [this PR](https://github.com/woocommerce/woocommerce-android/pull/1150). There were several new interface methods added existing classes:

**LoginAnalyticsListener**
- `trackConnectedSiteInfoRequested`
- `trackConnectedSiteInfoFailed`
- `trackConnectedSiteInfoSucceeded`

**LoginListener**
- `showHelpFindingConnectedEmail`
- `gotConnectedSiteInfo`

I've implemented these new interface methods in `LoginActivity` and `LoginAnalyticsTracker` but there is no logic needed since none of them are currently being used in WPAndroid. Those are the only two changes to WPAndroid directly. 

Full documentation around the changes to login library are documented in [this PR](https://github.com/woocommerce/woocommerce-android/pull/1150), but the **WPAndroid app should see no difference in the login experience**.

### Recommended Test Scenarios
- Verify login flow is unchanged from the current login flow of WPAndroid
- Verify the text in each of the login views is also unchanged
- Login with email
- Login by site url
- Login w/ Magic Link
- Login w/ Google

**Note: This PR is not ready for merge. [This PR](https://github.com/wordpress-mobile/WordPress-Login-Flow-Android/pull/22) must first be merged, and then a new branch/PR will be submitted with the fresh changes ready for final review.**

### CODE REVIEW

File: .idea/codeStyles/Project.xml [hunk 1]
```diff
@@ -1,42 +1,10 @@
 <component name="ProjectCodeStyleConfiguration">
```
Comment (malinajirka) [inline .idea/codeStyles/Project.xml]: Are these changes intentional?

File: .idea/inspectionProfiles/profiles_settings.xml [hunk 2]
```diff
@@ -1,7 +0,0 @@
-<component name="InspectionProjectProfileManager">
```
Comment (malinajirka) [inline .idea/inspectionProfiles/profiles_settings.xml]: Are these changes intentional?

File: .idea/codeStyles/Project.xml (hunk 1 above)
Comment (AmandaRiu) [reply .idea/codeStyles/Project.xml]: Good catch. Reverted in 3d43b3c2ca0d96045f697c1072cee483956de918

File: .idea/inspectionProfiles/profiles_settings.xml (hunk 2 above)
Comment (AmandaRiu) [reply .idea/inspectionProfiles/profiles_settings.xml]: Another good catch! Reverted in d8061682ba9f6d537e834712f013401b88a86f45

**Comment by AmandaRiu** [comment]
@malinajirka I've reverted the unintentional changes to the config files. Ready for another round 👍

**Comment by AmandaRiu** [comment]
Actually, I just re-read the instructions and I need to make a new branch. So ignore this for now. I'll create a new PR.
